assets/js/admin.js

```diff
@@ -198,15 +198,22 @@ function populateTemplate( item, dialog, action ) {
 					</form>
 					<promote-job-template>
 						<div slot="buttons" class="promote-buttons-group">
-								<a class="promote-button button button-primary" target="_blank" href="${ this.getAttribute( 'data-href' ) }">${ job_manager_admin_params.job_listing_promote_strings.promote_job }</a>
-								<a class="promote-button button button-secondary" target="_blank" href="#">${ job_manager_admin_params.job_listing_promote_strings.learn_more }</a>
+							<a id="wpjm-promote-button" class="promote-button button button-primary" target="_blank" rel="noopener noreferrer" href="${ this.getAttribute( 'data-href' ) }">${ job_manager_admin_params.job_listing_promote_strings.promote_job }</a>
+							<a class="promote-button button button-secondary" target="_blank" rel="noopener noreferrer" href="#">${ job_manager_admin_params.job_listing_promote_strings.learn_more }</a>
 						</div>
 					<promote-job-template>`;
 				}
+
 				if ( 'deactivate' === action ) {
 					let deactivateButton = dialog.querySelector( '.deactivate-promotion' );
 					deactivateButton.setAttribute( 'href', this.getAttribute( 'data-href' ) );
 				}
+
+				if ( 'promote' === action ) {
+					dialog.querySelector( '#wpjm-promote-button' ).addEventListener( 'click', function() {
+						dialog.close();
+					} );
+				}
```

Comment:
Could we maybe move this code to the `if ( 'promote' === action ) {` that we already have above? But still after the `innerHTML` of course? I don't see why we would have this duplicated condition based on `'promote' === action` here.. :thinking:

Reply:
Ah, good catch! I didn't notice that.

Updated in https://github.com/Automattic/WP-Job-Manager/pull/2508/commits/dfe229e979179974e8837fb481edad9cd04ff373. I'm going ahead with the merge since it was a small change, but feel free to comment anything else that I can fix in another PR.

Reply:
Good one! It is perfect now. Thanks!